components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -73,7 +73,24 @@ class ToolbarPresenter(
             Toolbar.SiteSecurity.INSECURE
         }
 
+        val session = tab.engineState.engineSession
+        if (session != null) {
+            try {
+                session.isIgnoredForTrackingProtection { offForThisSession ->
+                    setTrackingProtectionState(tab, offForThisSession)
+                }
```

Comment:
Having a callback here is problematic. Is that executed asynchronously? If so then there's no guarantee that at the time of executing the callback we are still "presenting" this state. :(

Reply:
> Is that executed asynchronously?

Unfortunately yes, we are making the [request to geckoview ](https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/ContentBlockingController.html#checkException-org.mozilla.geckoview.GeckoSession-) and they are returning us a `GeckoResult`. Should I make this call blocking?

```kotlin
    override fun isIgnoredForTrackingProtection(onResult: (Boolean) -> Unit) {
        runtime.contentBlockingController.checkException(geckoSession).accept {
            if (it != null) {
                onResult(it)
            } else {
                onResult(false)
            }
        }
    }
```

Reply:
Is there a good time where we could do this call and add it to the state? E.g. do we know when this changes (if it goes through our API) so that we can know when to update the state? Ideally the presenter only renders the state and has no side effects.

Reply:
>  E.g. do we know when this changes.

Yes, it goes through our API `TrackingProtectionUseCases.AddExceptionUseCase` https://github.com/mozilla-mobile/android-components/blob/b9c4b125b006532ad788560c8cad522d9a4fb9cd/components/feature/session/src/main/java/mozilla/components/feature/session/TrackingProtectionUseCases.kt#L40


I can add an additional property to `TrackingProtectionState` to indicate that is off for this site and change it when is removed `TrackingProtectionUseCases.RemoveExceptionUseCase`

https://github.com/mozilla-mobile/android-components/blob/b9c4b125b006532ad788560c8cad522d9a4fb9cd/components/feature/session/src/main/java/mozilla/components/feature/session/TrackingProtectionUseCases.kt#L61

Reply:
I updated the pr